### PR TITLE
feat(server): integration GraphQL schema

### PR DESCRIPTION
**Pull request by yk-eukarya**
# Overview

### CODE REVIEW

File: server/schemas/integration.graphql [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  description: String
+  logoUrl: URL!
+  type: IntegrationType!
+  token: String!
+  developerId: ID!
+  developer: User!
+  webhook: [Webhook!]!
+  createdAt: DateTime!
+  updatedAt: DateTime!
+}
```
Comment (rot1024) [inline server/schemas/integration.graphql:18]:
```suggestion
type Integration implements Node {
  id: ID!
  name: String!
  description: String
  logoUrl: URL!
  type: IntegrationType!
  developerId: ID!
  developer: User!
  createdAt: DateTime!
  updatedAt: DateTime!
  config: IntegrationConfig # only for a developer of this integration
}

type IntegrationConfig {
  token: String!
  webhooks: [Webhook!]!
}
```

- I have divided the information into different types that should only be visible to the developer himself.
- "type" is not usable as a field name?

File: server/schemas/integration.graphql [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  integration: Integration!
+}
+
+type DeleteIntegrationPayload {
+  integrationId: ID!
+}
+
+extend type Query {
+  integrations(type: IntegrationType, keyword: String): [IntegrationPayload]!
+}
```
Comment (rot1024) [inline server/schemas/integration.graphql:50]:
```suggestion
```

I think it is not needed. Developers themselves add integrations directly to the workspace they belong to. No search function is required.

File: server/schemas/integration.graphql [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+extend type Mutation {
+  createIntegration(input: CreateIntegrationInput!): IntegrationPayload
+  updateIntegration(input: UpdateIntegrationInput!): IntegrationPayload
+  deleteIntegration(input: DeleteIntegrationInput!): DeleteIntegrationPayload
```
Comment (rot1024) [inline server/schemas/integration.graphql:55]: To install an integration to the workspace, just we can invite it as a member of the workspace. So there is no need to add a new mutation to install it. Is it right?

File: server/schemas/integration_connection.graphql [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+extend type Query {
+  integrationConnections(keyword: String, pageInfo: Pagination!): [IntegrationConnectionConnection]!
+}
+
+extend type Mutation {
+  createIntegrationConnection(input: CreateIntegrationConnectionInput!): Boolean
+  updateIntegrationConnection(input: UpdateIntegrationConnectionInput!): Boolean
+  deleteIntegrationConnection(input: DeleteIntegrationConnectionInput!): Boolean
```
Comment (rot1024) [inline server/schemas/integration_connection.graphql:48]: `integration_connection.graphql` is not needed.

A connection is an object that enables paging according to [Relay Cursor Pagination](https://relay.dev/graphql/connections.htm), not something the user creates.

File: server/schemas/integration_webhook.graphql [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+type Webhook implements Node {
+  id: ID!
+  name: String!
+  url: URL!
+  active: Boolean!
+  event: [WebhookEvent!]!
+  createdAt: DateTime!
+  updatedAt: DateTime!
+}
```
Comment (rot1024) [inline server/schemas/integration_webhook.graphql:17]:
```suggestion
type Webhook {
  id: ID!
  name: String!
  url: URL!
  active: Boolean!
  trigger: WebhookTrigger!
  createdAt: DateTime!
  updatedAt: DateTime!
}

type WebhookTrigger {
  onItemCreate: Boolean!
  onItemUpdate: Boolean!
  onItemDelete: Boolean!
  onAssetUpload: Boolean!
  onAssetDeleted: Boolean!
  onItemPublish: Boolean!
  onItemUnpublish: Boolean!
}
```

Webook is an entity, but it is better not to be a root entity. It just belongs to an Integration.

File: server/schemas/integration_webhook.graphql [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+# Inputs
+
+input WebhookEventInput {
+  model: ID!
+  onCreate: Boolean!
+  onUpdate: Boolean!
+  onDelete: Boolean!
+  onApiAccess: Boolean!
+}
```
Comment (rot1024) [inline server/schemas/integration_webhook.graphql:27]:
```suggestion
input WebhookTriggerInput {
  onItemCreate: Boolean
  onItemUpdate: Boolean
  onItemDelete: Boolean
  onAssetUpload: Boolean
  onAssetDeleted: Boolean
  onItemPublish: Boolean
  onItemUnpublish: Boolean
}
```

File: server/schemas/integration_webhook.graphql [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+input CreateWebhookInput {
+  name: String!
+  url: URL!
+  active: Boolean!
+  event: [WebhookEventInput!]!
```
Comment (rot1024) [inline server/schemas/integration_webhook.graphql:33]:
```suggestion
  trigger: WebookTrigger
```

File: server/schemas/integration_webhook.graphql [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+input UpdateWebhookInput {
+  webhookId: ID!
+  name: String
+  url: URL
+  active: Boolean
+  event: [WebhookEventInput!]
```
Comment (rot1024) [inline server/schemas/integration_webhook.graphql:41]:
```suggestion
  trigger: WebookTrigger
```

**Review by rot1024 — CHANGES_REQUESTED** [review]

File: server/schemas/integration.graphql (hunk 2 above)
Comment (yk-eukarya) [reply server/schemas/integration.graphql:50]: we can add the integrations to the `Me` query but how about, the other users want to list the integration to install them?

File: server/schemas/integration.graphql (hunk 3 above)
Comment (yk-eukarya) [reply server/schemas/integration.graphql:55]: Yes sure

File: server/schemas/integration.graphql (hunk 2 above)
Comment (rot1024) [reply server/schemas/integration.graphql:50]: No need. Only private integrations are available at the beginning. Public integrations will not be implemented at least this year.

File: server/schemas/integration.graphql [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+type Integration implements Node {
+  id: ID!
+  name: String!
+  description: String
+  logoUrl: URL!
+  iType: IntegrationType!
+  developerId: ID!
+  developer: User!
+  config: IntegrationConfig # only for a developer of this integration
```
Comment (rot1024) [inline server/schemas/integration.graphql:19]:
```suggestion
  # only for a developer of this integration
  config: IntegrationConfig
```

**Review by rot1024 — APPROVED** [review]

**Pull request merged**